src/fabric_cicd/_items/_lakehouse.py

```diff
@@ -145,11 +142,21 @@ def publish_shortcuts(fabric_workspace_obj: FabricWorkspace, item_obj: Item, sho
     """
     for shortcut in shortcut_dict.values():
         # https://learn.microsoft.com/en-us/rest/api/fabric/core/onelake-shortcuts/create-shortcut
-        fabric_workspace_obj.endpoint.invoke(
-            method="POST",
-            url=f"{fabric_workspace_obj.base_api_url}/items/{item_obj.guid}/shortcuts?shortcutConflictPolicy=CreateOrOverwrite",
-            body=shortcut,
-        )
+        try:
+            fabric_workspace_obj.endpoint.invoke(
+                method="POST",
+                url=f"{fabric_workspace_obj.base_api_url}/items/{item_obj.guid}/shortcuts?shortcutConflictPolicy=CreateOrOverwrite",
+                body=shortcut,
+            )
+            logger.info(f"{constants.INDENT}{shortcut['name']} Shortcut Published")
+        except Exception as e:
+            if "continue_on_shortcut_failure" in constants.FEATURE_FLAG:
+                logger.error(
+                    "Failed to publish shortcut. This usually happens when the lakehouse containing the source for this shortcut is published as a shell and has no data yet."
+                )
+                logger.info("The publish process will continue with the other items.")
+                break
+            raise e
```

Comment:
Not sure, let's check if one of our custom errors is suitable here from the exceptions class in _common module?